indico/modules/events/registration/badges.py

```diff
@@ -99,6 +101,23 @@ def _draw_badge(self, canvas, registration, template, tpl_data, pos_x, pos_y):
             else:
                 continue
 
+            updates = {
+                'item': item,
+                'items': items,
```

Comment:
Do you plan to modify these? If not I'd keep those outside and rather make the registration the sender and pass item (and items if needed) as separate kwargs. And then only keep the "simple" values like height and positions in the dict.

But if you could explain how you intend to use the signal it'll certainl be easier to see what would be the best approach here...

Reply:
We are modifying only the `item` and the `text`

We are using the signal to change some of the item position and color if it overlaps with other items. And also change the item color based on the registration badge attributes.

Reply:
I would go for something like this then:

```python
            item_data = {'item': item, 'text': text, 'pos_x': pos_x, 'pos_y': pos_y}
            for update in values_from_signal(
                signals.event.designer.draw_item_on_badge.send(template, registration=registration, items=items, 
                                                               height=self.height, width=self.width, data=item_data),
                as_list=True
            ):
                item_data.update(update)
            self._draw_item(canvas, item_data['item'], tpl_data, item_data['text'], item_data['pos_x'],
                            item_data['pos_y'])
```

That way you keep the updateable data separate from the rest, and use a more logical sender (I don't have a strong opinion whether the registration or the template should be the sender BTW).

Reply:
Okay, I'll add these changes and use the registration as the sender. Thank you!